specification/mcpp/resource-manager/Microsoft.CloudPartnerProgramMembership/preview/2023-02-09-preview/mcpp.json

```diff
@@ -0,0 +1,447 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": ["CloudPartnerProgramMemberships"],
+        "operationId": "CloudPartnerProgramMemberships_CreatePurchase",
+        "x-ms-examples": {
+          "putOperations": { "$ref": "./examples/createMcppPurchase.json" }
+        },
+        "description": "Create a CloudPartnerProgramMembership purchase",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ProductCodeParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List the CloudPartnerProgramMembership purchase details .",
+            "schema": {
+              "$ref": "#/definitions/McppPurchaseDetailsResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["CloudPartnerProgramMemberships"],
+        "operationId": "CloudPartnerProgramMemberships_CancelMcppPuchase",
+        "x-ms-examples": {
+          "deleteOperations": { "$ref": "./examples/cancelMcppPurchase.json" }
+        },
+        "description": "The operation to cancel a CloudPartnerProgramMembership offer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Canceled the purchase"
+          },
+          "204": {
+            "description": "purchase record does not exist"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "McppPurchaseDetailsResponse":{
+      "description": "MCPP Purchase details response",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
```

Comment:
>properties [](http://example.com/codeflow?start=7&length=10)

you can leverage the proxyresource definition from the common types

Reply:
Sorry, I didn't get this.  Where / how can i leverage this ?